django/contrib/postgres/fields/array.py

```diff
@@ -288,3 +289,13 @@ def __init__(self, start, end):
 
     def __call__(self, *args, **kwargs):
         return SliceTransform(self.start, self.end, *args, **kwargs)
+
+
+@CombinedExpression.dispatch(CombinedExpression.ADD, ArrayField, None)
+@CombinedExpression.dispatch(CombinedExpression.ADD, None, ArrayField)
+@CombinedExpression.dispatch(CombinedExpression.ADD, ArrayField, ArrayField)
```

Comment:
Also, `ArrayField + ArrayField` is ambiguous wrt `ArrayField + Field`, so we need all 3 dispatches. 
There might be a way to avoid raising warnings if all dispatches point to the same implementation. I'll give it a shot.